src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,169 @@
+package parser;
+
+import command.*;
+import dukeexceptions.*;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    public Parser() {}
+    public enum Commands {
+        BYE,
+        LIST,
+        UNMARK,
+        MARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        ADD,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * Takes Command Input and returns a command object with the relevant execute code
+     * @param cmdInput
+     * @return a command object
+     * @throws InvalidCmd For when command is not recognised
+     * @throws InvalidDateTimeFormat for when the user gives the date time in an incorrect format
+     * @throws EventEmptyException for when any of the fields of the event cmd is empty
+     * @throws DeadlineEmptyException for when any of the fields of the deadline cmd is empty
+     */
+    public Command parseCommand(String cmdInput)
+            throws InvalidCmd, InvalidDateTimeFormat, EventEmptyException, DeadlineEmptyException {
+        Commands cmd;
+        try {
+            cmd = Commands.valueOf(cmdInput.split(" ", 2)[0].toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new InvalidCmd(cmdInput);
+        }
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+        switch (cmd) {
+            case BYE:
+                return new ByeCmd();
+            case LIST:
+                return new ListCmd();
+            case UNMARK:
+                return new UnmarkCmd(Integer.parseInt(cmdInput.substring(7)));
+            case MARK:
+                return new MarkCmd(Integer.parseInt(cmdInput.substring(5)));
+            case TODO:
+                return new TodoCmd(cmdInput.substring(5));
+            case EVENT:
+                String[] inputs = parseEvent(cmdInput);
+                try {
+                    return new EventCmd(inputs[0],
+                            LocalDateTime.parse(inputs[1].substring(5), formatter),
+                            LocalDateTime.parse(inputs[2].substring(3), formatter));
```

Comment:
I like how you are cautions of the character limit for each line and inserted line breaks here.